### PR TITLE
Header and Footer refactoring

### PR DESCRIPTION
**Pull request by stefaniacardenas**
# Problem

At the moment the JS SDK displays the "powered by Onfido" logo at the bottom of each screen, as it was a footer. The logo is not enclosed in a footer div and is currently just a background image within the main container `.onfido-sdk-ui-Modal-inner`, this means that the logo is not a DOM element in itself and therefore it doesn't occupy any space, resulting in other elements often overriding or covering it.

This is an issue when implementing new screens as we have to be extra careful not to cover the "powered by Onfido" logo and it creates inconsistency in the UI of different screens. 
This is also an issue when the content needs to be spread evenly between the header and the footer or when we need to use flexbox.

# Solution

The "powered by Onfido" logo  should be enclosed in a footer that is placed at the bottom of each screens. It should work on the following scenarios:

* Standard Mode, where the content is contained between the header and the footer, ie on Desktop where the main container is smaller than the viewport
* Scrollable Mode, where content needs to be scrollable, ie document select screen on small screens
* Full Screen mode, where footer and back button will be displayed as an overlay

The Full Screen mode will be implemented as part of the Live Capture feature. For now, this mode can be tested as a dummy screen by adding `?liveCapture=true` to the url.

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [n/a] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [n/a] Have new automated tests been implemented?
- [n/a] Have new manual tests been written down?
- [x] Have tests passed locally?
- [n/a] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-329-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Capture/index.js [hunk 1]
```diff
@@ -16,17 +16,19 @@ const BackDocumentCapture = options => <DocumentCapture {...options} />
...
+const FaceCapture = appendToTracking(options =>
+  <Capture autoCapture={false} {...options} />, 'capture')
 
 FaceCapture.defaultProps = {
   useWebcam: true,
   method: 'face',
   side: null
 }
 
+FaceCapture.isFullScreen = true
```
Comment (rfreitas) [inline src/components/Capture/index.js]: is this still necessary then?

File: src/components/Capture/capture.js [hunk 2]
```diff
@@ -201,8 +200,9 @@ class Capture extends Component {
     this.setState({error: null})
   }
 
-  render ({useWebcam, back, i18n, termsAccepted, ...other}) {
+  render ({useWebcam, back, i18n, termsAccepted, useFullScreen, ...other}) {
     const useCapture = (!this.state.uploadFallback && useWebcam && isDesktop && this.state.hasWebcam)
+    useFullScreen(useCapture)
```
Comment (rfreitas) [inline src/components/Capture/capture.js]: any danger with calling this on `render`, can it create an infinite recursion?

File: src/components/Camera/style.css [hunk 3]
```diff
@@ -1,9 +1,28 @@
...
 .video-overlay {
   /* Relatively positioned so absolute children are bound to this overlay */
   position: relative;
   overflow: hidden;
-  margin-top: 16px;
-  width: 100%;
+  margin: 0 40px 16px;
+}
+
+.fullScreen {
```
Comment (rfreitas) [inline src/components/Camera/style.css]: is full screen layouting done at step level too? can it not be done at container level only?

File: src/components/NavigationBar/style.css [hunk 4]
```diff
@@ -1,13 +1,28 @@
...
   @media (--small-viewport) {
-    margin-left: 12px;
-    margin-top: -32px;
+    padding-left: 12px;
+  }
+}
+
+.fullScreenNav {
+  position: absolute;
+  z-index: 3;
```
Comment (rfreitas) [inline src/components/NavigationBar/style.css]: do we need the z-index? can we not use the dom order instead?

File: src/components/PrivacyStatement/style.css [hunk 5]
```diff
@@ -1,18 +1,10 @@
 @import (less) "../Theme/constants.css";
 
 .privacy {
-  /*
```
Comment (rfreitas) [inline src/components/PrivacyStatement/style.css]: nice!

File: src/components/Router/StepsRouter.js [hunk 6]
```diff
@@ -18,16 +21,26 @@ class StepsRouter extends Component {
 
   currentComponent = () => this.props.componentsList[this.props.step]
 
+  useFullScreen = (value) => this.setState({fullScreen: value})
+
+  hasLiveCaptureOption = (options) => options && options.liveCapture
```
Comment (rfreitas) [inline src/components/Router/StepsRouter.js]: is this needed?

File: src/index.ejs [hunk 7]
```diff
@@ -40,6 +40,7 @@
                       return a;
                     }, {});
       var useModal = options.useModal === "true"
+      var liveCapture = options.liveCapture === "true"
```
Comment (rfreitas) [inline src/index.ejs]: is this needed?

File: src/components/Capture/index.js (hunk 1 above)
Comment (stefaniacardenas) [reply src/components/Capture/index.js]: Yes, it is. Otherwise the full screen options will be applied to the confirm screen of the face step too.

Comment (rfreitas) [reply src/components/Capture/index.js]: if we are using the callback, let's just do that.

File: src/index.ejs (hunk 7 above)
Comment (stefaniacardenas) [reply src/index.ejs]: This is test code that I have added to facilitate acceptance. Will get rid of it.

File: src/components/Router/StepsRouter.js (hunk 6 above)
Comment (stefaniacardenas) [reply src/components/Router/StepsRouter.js]: This is test code that I have added to facilitate acceptance. Will get rid of it.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://329-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Footer/index.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { h } from 'preact'
+import style from './style.css'
+import footerImg from './assets/powered-by-onfido.svg'
```
Comment (rfreitas) [inline src/components/Footer/index.js]: good news that it works on IE11 !

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-329-pr-onfido-sdk-ui-onfido.surge.sh

File: package.json [hunk 9]
```diff
@@ -77,6 +77,7 @@
     "style-loader": "^0.13.0",
     "superstatic": "^4.0.2",
     "surge": "^0.18.0",
+    "svg-url-loader": "^2.3.2",
```
Comment (rfreitas) [inline package.json]: no needed, right?

File: src/components/Confirm/style.css [hunk 10]
```diff
@@ -47,6 +35,10 @@
   border-radius: 3px;
 }
 
+.pdfImage {
```
Comment (rfreitas) [inline src/components/Confirm/style.css]: who else is using `.image`? why not just add `margin: 0` to it?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://329-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-329-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**